consensus/nodes/create_nodes.py

```diff
@@ -17,9 +17,9 @@ def get_random_provider_using_weights(defaults):
     # remove providers if no api key
     provider_weights = defaults['provider_weights']
     default_value = '<add_your_api_key_here>'
-    if 'OPENAIKEY' not in os.environ or os.environ['OPENAIKEY'] == default_value:
+    if ('openai' in provider_weights and provider_weights['openai'] > 0 and ('OPENAIKEY' not in os.environ or os.environ['OPENAIKEY'] == default_value)):
         provider_weights.pop('openai')
-    if 'HEURISTAIAPIKEY' not in os.environ or os.environ['HEURISTAIAPIKEY'] == default_value:
+    if ('heuristai' in provider_weights and provider_weights['heuristai'] > 0 and('HEURISTAIAPIKEY' not in os.environ or os.environ['HEURISTAIAPIKEY'] == default_value)):
```

Comment:
`provider_weights['heuristai'] > 0` I think we are making the if; overly complex and less legible. Let the random weights function handle this.